### PR TITLE
extended camera example

### PR DESCRIPTION
**Pull request by scottlamb**
I find it useful to have a more extended example that works easily
out of the box. In particular, one that:

* supports authentication for everything. My cameras require it for
  almost all operations, so eg get_stream_uri didn't work.
* uses real argument parsing so usage is more obvious.
* automatically finds the right URIs for the different SOAP services.
  Also unlike the README examples, my cameras don't let you send
  everything to the root path.
* supports a couple longer operations. They're admittedly a little
  arbitrary.

Not everything works as of this commit, due to media-io/yaserde#76
and at least one other problem.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -37,19 +37,16 @@ To [discover](onvif-rs/examples/discovery.rs) devices on the local network:
...
 ```shell script
-cargo run --example camera_datetime -- http://192.168.0.2:8000
-```
+cargo run --example camera -- help
 
-To [set camera hostname](onvif-rs/examples/camera_set_hostname.rs) using credentials:
-```shell script
-cargo run --example camera_set_hostname -- http://192.168.0.2:8000 admin qwerty cam2
-```
+cargo run --example camera -- get_datetime --uri=http://192.168.0.2:8000
```
Comment (DmitrySamoylov) [inline README.md]: Looks like a typo, think the correct command is `get-system-date-and-time`

File: README.md [hunk 2]
```diff
@@ -37,19 +37,16 @@ To [discover](onvif-rs/examples/discovery.rs) devices on the local network:
...
 
-To [get RTSP stream URL](onvif-rs/examples/camera_get_stream_uri.rs):
-```shell script
-cargo run --example get_stream_uri -- http://192.168.0.2:8000
+cargo run --example camera -- set_hostname \
```
Comment (DmitrySamoylov) [inline README.md]: `set-hostname`

File: README.md [hunk 3]
```diff
@@ -37,19 +37,16 @@ To [discover](onvif-rs/examples/discovery.rs) devices on the local network:
...
+    --uri=http://192.168.0.2:8000 --username=admin --password=qwerty cam2
+
+cargo run --example camera -- get_stream_uris --uri=http://192.168.0.2:8000
```
Comment (DmitrySamoylov) [inline README.md]: `get-stream-uris`

**Review by DmitrySamoylov — CHANGES_REQUESTED** [review]

**Review by zeenix — APPROVED** [review]
Except for what @DmitrySamoylov pointed out, LGTM.

**Comment by scottlamb** [comment]
Thanks! Fixed the README errors you pointed out. I saw zeenix's fe15362 so I'm trying out this fixup commit thing.

**Review by DmitrySamoylov — APPROVED** [review]

**Comment by DmitrySamoylov** [comment]
Nice work, thanks! Now you can run `git rebase -i --autosquash` and force-push

**Review by victor-soloviev — APPROVED** [review]

**Comment by scottlamb** [comment]
Rebased.

**Pull request merged**